alloc.c

```diff
@@ -662,22 +662,27 @@ bool liftoff_output_apply(struct liftoff_output *output, drmModeAtomicReq *req)
 		    "score=%d:", (void *)output, result.best_score);
 
 	/* Apply the best allocation */
-	i = 0;
+	liftoff_log(LIFTOFF_DEBUG, "Final assignment of layers to planes:");
+	i = j = 0;
 	liftoff_list_for_each(plane, &device->planes, link) {
 		layer = result.best[i];
 		i++;
 		if (layer == NULL) {
 			continue;
 		}
 
-		liftoff_log(LIFTOFF_DEBUG, "  Layer %p -> plane %"PRIu32,
-			    (void *)layer, plane->id);
+		j++;
+		liftoff_log(LIFTOFF_DEBUG, "  [%zu] Layer %p -> plane %"PRIu32
```

Comment:
Does the `j` counter provide any useful information?

Reply:
It just indexes the assignments. The question is if in the log at the end we should not compare `j>0`.

Reply:
Ah, yeah, printing a message when no assignments are established seems like a good idea.